cpp/src/groupby/hash/groupby.cu

```diff
@@ -667,10 +654,18 @@ std::pair<std::unique_ptr<table>, std::vector<aggregation_result>> groupby(
   rmm::cuda_stream_view stream,
   rmm::mr::device_memory_resource* mr)
 {
+  auto const has_list_column =
+    std::any_of(keys.begin(), keys.end(), [](cudf::column_view const& col) {
+      return col.type().id() == type_id::LIST;
+    });
+  if (has_list_column and include_null_keys == cudf::null_policy::EXCLUDE) {
+    CUDF_FAIL("Null LIST keys cannot be excluded.");
+  }
+
```

Comment:
This function is called for hash-based aggregates. As I know, nested types are not supported yet at this function.

Reply:
Line 634-636 above are not enough: both struct and list types are not supported.

Reply:
So supporting nested types as values (not keys) will be another similar PR :smiley:

Reply:
The PR title should also be updated to "Update `groupby` to use new row operators for keys".